lib/routes/smzdm/product.ts

```diff
@@ -0,0 +1,81 @@
+import { Route, DataItem } from '@/types';
+import { ofetch } from 'ofetch';
+import { load } from 'cheerio';
+
+export const route: Route = {
+    path: '/product/:id',
+    categories: ['shopping'],
+    example: '/smzdm/product/zm5vzpe',
+    parameters: { id: '商品 id，网址上直接可以看到' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['https://wiki.smzdm.com/p/:id'],
+            target: '/product/:id',
+        },
+    ],
+    name: '商品',
+    maintainers: ['chesha1'],
+    handler,
+};
+
+async function handler(ctx) {
+    const link = `https://wiki.smzdm.com/p/${ctx.req.param('id')}`;
+
+    const response = await ofetch(link);
+    const $ = load(response);
+    const title = $('title').text();
+
+    const items: DataItem[] = [];
+
+    // get simple info from list
+    $('ul#feed-main-list li').each(function () {
+        const altText = $(this).find('img').attr('alt');
+        const link = $(this).find('h5.feed-block-title a').attr('href');
+        const price = $(this).find('.z-highlight').text();
+        const title = altText + ' ' + price;
+        const description = $(this).find('.feed-block-descripe').text().replaceAll(/\s+/g, '');
+        const item = {
+            title,
+            link,
+            description,
+        };
+        items.push(item);
+    });
+
+    // get detail info from each item
+    const out = await Promise.all(
+        items.map(async (item) => {
+            const response = await ofetch(item.link as string);
+            const $ = load(response);
+
+            // filter outdated articles
+            if ($('span.old').length > 0) {
+                return null;
+            } else {
+                const pubDate = $('meta[name="weibo:webpage:create_at"]').attr('content');
+                item.pubDate = pubDate;
+
+                if (item.description === '阅读全文') {
+                    item.description = $('p[itemprop="description"]').first().html() as string;
+                }
+
+                return item;
+            }
+        })
+    );
```

Comment:
Please [cache](https://docs.rsshub.app/joinus/advanced/use-cache) the returned result.

Reply:
On the RSSHub server side, what is the cache expiration period?

Also, what types of content should be cached? While browsing the source code of other routes, I noticed that some content is cached while other content is not. I'm wondering if there's a clear guideline for this.